### ISSUE TITLE
Exe logo in task manager is strected for win-ui 3 application.

### ISSUE THREAD

**Issue by Mohan-Babblu** [microsoft/microsoft-ui-xaml#8134]
### Describe the bug

Hi , I am currently working in an Win-Ui 3 Project , where i have added the icon to my application , but the thing is while runnning the application in task manager , the application icon is strected out  and not displaying properly .

### Steps to reproduce the bug

1.Create a winui-3 project.
2.In mainui.csproj add "<ApplicationIcon>Assets\Test.ico</ApplicationIcon>".
3.In mainwindow.xaml.cs class   "AppWindow _appWindow;" , & In constructor "_appWindow.SetIcon(@"Assets\\Test.ico");".
4.Run The application.
5.Open Task Manger and expand the application. 

### Expected behavior

Logo should display without strecthing in task manager.

### NuGet package version

WinUI 3 - Windows App SDK 1.2.3: 1.2.230118.102

### Windows version

Windows 11 (21H2): Build 22000

### Additional context

Image resoultion , i've added 256*256 , I ve changed the resolutions and tested  not working as expected. 
Note: In some system the logo is coming properly as expected .

**Comment by Mohan-Babblu** [comment]
![Issue Image](https://user-images.githubusercontent.com/88727251/215397989-044b683d-1052-49e0-ad2b-11b82a77fab7.png)

**Comment by Mohan-Babblu** [comment]
@JaiganeshKumaran , This will make the icons in titlebar and taskbar faded.

**Comment by castorix** [comment]
I tested with a round icon containing 3 formats (16x16, 32x32, 48x48) and it is OK (Windows 10 22H2)  =>

Task Manager :
![TaskManager_ico](https://user-images.githubusercontent.com/22345506/215537656-3ff9f742-f1be-47e9-b273-9beb2145a76c.jpg)
Taskbar :
![Taskbar_icon_test](https://user-images.githubusercontent.com/22345506/215537710-6358ba61-3f99-4bd9-97e0-8d7bc4bebb0f.jpg)
Caption :
![Caption_Icon](https://user-images.githubusercontent.com/22345506/215537743-2f61c992-54f0-4718-a25a-21fec33c1363.jpg)

**Comment by Mohan-Babblu** [comment]
@castorix 

> I tested with a round icon containing 3 formats (16x16, 32x32, 48x48) and it is OK (Windows 10 22H2) =>
> 
> Task Manager : ![TaskManager_ico](https://user-images.githubusercontent.com/22345506/215537656-3ff9f742-f1be-47e9-b273-9beb2145a76c.jpg) Taskbar : ![Taskbar_icon_test](https://user-images.githubusercontent.com/22345506/215537710-6358ba61-3f99-4bd9-97e0-8d7bc4bebb0f.jpg) Caption : ![Caption_Icon](https://user-images.githubusercontent.com/22345506/215537743-2f61c992-54f0-4718-a25a-21fec33c1363.jpg)

In WIndows 10 u wont find the difference for any resolution images , In windows 11 image will strect out .

**Comment by Mohan-Babblu** [comment]
> Can you try making a 16*16 version of the icon?

If i make that as 16*16 or any In taskbar the logo is looking faded  & Still i can see the  logo in task manager Strected , Is there anyway to override the SetIcon property to differentiate task bar and task manager.

**Comment by castorix** [comment]
On Windows 10, for the window icon, I get correct results with 
_butterfly.ico_ at https://github.com/castorix/WinUI3_NotifyIcon/tree/master/Assets
(contains 16x16, 32x32, 48x48, 128x128)
and

Icons :

        private IntPtr m_hSmallIcon = IntPtr.Zero;
        private IntPtr m_hBigIcon = IntPtr.Zero;

Test :

                m_hSmallIcon = LoadImage(IntPtr.Zero, @".\Assets\Butterfly.ico", IMAGE_ICON, 16, 16, LR_LOADFROMFILE);
                m_hBigIcon = LoadImage(IntPtr.Zero, @".\Assets\Butterfly.ico", IMAGE_ICON, 32, 32, LR_LOADFROMFILE);
                IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
                SendMessage(hWnd, WM_SETICON, (IntPtr)ICON_SMALL, m_hSmallIcon);
                SendMessage(hWnd, WM_SETICON, (IntPtr)ICON_BIG, m_hBigIcon);

  Declarations :              
                
        [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
        public static extern IntPtr LoadImage(IntPtr hInst, string lpszName, UInt32 uType, int cxDesired, int cyDesired, UInt32 fuLoad);

        public const int IMAGE_BITMAP = 0;
        public const int IMAGE_ICON = 1;
        public const int IMAGE_CURSOR = 2;
        public const int IMAGE_ENHMETAFILE = 3;

        public const int LR_DEFAULTCOLOR = 0x00000000;
        public const int LR_MONOCHROME = 0x00000001;
        public const int LR_COLOR = 0x00000002;
        public const int LR_COPYRETURNORG = 0x00000004;
        public const int LR_COPYDELETEORG = 0x00000008;
        public const int LR_LOADFROMFILE = 0x00000010;

        public const int ICON_SMALL = 0;
        public const int ICON_BIG = 1;
        public const int ICON_SMALL2   = 2;

        [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
        public static extern int SendMessage(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);

        public const int WM_GETICON = 0x007F;
        public const int WM_SETICON = 0x0080;

**Comment by Mohan-Babblu** [comment]
> On Windows 10, for the window icon, I get correct results with _butterfly.ico_ at https://github.com/castorix/WinUI3_NotifyIcon/tree/master/Assets (contains 16x16, 32x32, 48x48, 128x128) and
> 
> Icons :
> 
> ```
>     private IntPtr m_hSmallIcon = IntPtr.Zero;
>     private IntPtr m_hBigIcon = IntPtr.Zero;
> ```
> 
> Test :
> 
> ```
>             m_hSmallIcon = LoadImage(IntPtr.Zero, @".\Assets\Butterfly.ico", IMAGE_ICON, 16, 16, LR_LOADFROMFILE);
>             m_hBigIcon = LoadImage(IntPtr.Zero, @".\Assets\Butterfly.ico", IMAGE_ICON, 32, 32, LR_LOADFROMFILE);
>             IntPtr hWnd = WinRT.Interop.WindowNative.GetWindowHandle(this);
>             SendMessage(hWnd, WM_SETICON, (IntPtr)ICON_SMALL, m_hSmallIcon);
>             SendMessage(hWnd, WM_SETICON, (IntPtr)ICON_BIG, m_hBigIcon);
> ```
> 
> Declarations :
> 
> ```
>     [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
>     public static extern IntPtr LoadImage(IntPtr hInst, string lpszName, UInt32 uType, int cxDesired, int cyDesired, UInt32 fuLoad);
> 
>     public const int IMAGE_BITMAP = 0;
>     public const int IMAGE_ICON = 1;
>     public const int IMAGE_CURSOR = 2;
>     public const int IMAGE_ENHMETAFILE = 3;
> 
>     public const int LR_DEFAULTCOLOR = 0x00000000;
>     public const int LR_MONOCHROME = 0x00000001;
>     public const int LR_COLOR = 0x00000002;
>     public const int LR_COPYRETURNORG = 0x00000004;
>     public const int LR_COPYDELETEORG = 0x00000008;
>     public const int LR_LOADFROMFILE = 0x00000010;
> 
>     public const int ICON_SMALL = 0;
>     public const int ICON_BIG = 1;
>     public const int ICON_SMALL2   = 2;
> 
>     [DllImport("User32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
>     public static extern int SendMessage(IntPtr hWnd, uint msg, IntPtr wParam, IntPtr lParam);
> 
>     public const int WM_GETICON = 0x007F;
>     public const int WM_SETICON = 0x0080;
> ```

In Windows 10 for all the resolution images u won't find the stretched-out icon in task manager but u can see the icon clarity variations, But in Windows 11 you can see the stretched-out logo in task manager and clarity variation according to the icon resolution.

**Comment by castorix** [comment]
You say that you get a stretched butterfly if you test with ICON_SMALL + ICON_BIG
with 16x16 + 32x32 or 
 32x32 + 48x48 or
 any other tests ?

**Comment by tranhuynhthaitrung** [comment]
@castorix 
I applied your solution. It work well with scale 100%
However, if we adjust scale to ?100%, we see that app icon will be smaller in task manager
P/s: This issue only happen on some machines

**Comment by YexuanXiao** [comment]
I'm not sure if this is a WindowsAppSDK bug or a Windows bug, but on my computer it unfortunately happened and the icons were stretched incorrectly.

**Comment by Leonard-The-Wise** [comment]
This is what I did to solve my problem (Windows 11, same thing: stretched "internal" icon on task manager):

- Followed @castorix solution to set icons with Win32 API SendMessage;
- The inner icon of task manager is now empty.
- I created a common resource file called 'appicon.rc' on my project root so I could embbed the icon directly on the program executable. The resource only contained one single line:
```.rc
IDI_APPICON ICON "Assets\\AppIcon.ico"
```
- Now I edited the .csproj and put a compilation action to the resource file:
```XML
  <ItemGroup>
	<ResourceCompile Include="appicon.rc" />
  </ItemGroup>
```

Then rebuild application.

Now task manager shows the icon unstretched.

PS: I think it's a bug with WInUI 3.
